### ISSUE TITLE
Review

### ISSUE THREAD

**Issue by Gpir0** [33Domenico/CI2024_lab1#2]
You've done a good job testing various algorithms and writing clean code. The RHCM v2 algorithm you have chosen takes an intelligent approach: selecting the set with the greatest coverage and excluding it from tweaks. The idea of a dynamic buffer that adjusts the algorithm's strength based on how "stuck" it is in finding a better solution is brilliant. It resolves many issues found in other proposed algorithms (including mine). As for the results, you’ve achieved an improvement compared to the classic hill climbing algorithm. For some instances, you achieved values above the student average, while for others, you were at the average.
<br>

A possible improvement would be to define a ranking of the set coverage and use that directly, instead of searching for the maximum in each tweak. Since the evaluation considers the number of calls to the "cost function", this improvement is not strictly necessary.
<br>

NOTE: I noticed that sometimes the graphical representation includes the maximum value of non-acceptable solutions (see attached photo.).  To avoid this, you could use history.append() only for only for valid solutions.
<br>
![issue](https://github.com/user-attachments/assets/9be174d0-3beb-45f5-8918-d907230b1310)

<br>

In conclusion, I believe your work is solid, especially given that the algorithm has been refined through previous attempts.
